lib/main/lang/projectService.ts

```diff
@@ -162,7 +163,13 @@ declare module "${modulePath}"{
         fs.writeFileSync(defLocation, joinedDtsCode);
     }
 
+    let tsFilesWithInvalidEmit = outputs
+        .filter((o) => o.emitError) // only select outputs that produced an emit error
+        .map((o) => o.errors.map((e) => e.filePath)) // extract file paths from errors
+        .reduce((a, b) => a.concat(b), []); // flatten the array
+
```

Comment:
Sorry, I don't quite understand what you mean by adding `sourceFileName` to `EmitOutput`